src/test-run/index.js

```diff
@@ -43,7 +43,8 @@ export default class TestRun extends Session {
         this.injectable.scripts.push('/testcafe-driver.js');
         this.injectable.styles.push('/testcafe-ui-styles.css');
 
-        this.errs = [];
+        this.errs                = [];
+        this.driverResponseCache = {};
```

Comment:
In fact we should keep just one last response in the cache. Driver can send repeated statuses just in a raw (it's not possible to get the following queue of statuses: `{ id: 'status1' }, { id: 'status2' }, { id: 'status1' }`)
Maybe it be better to use a simple variable for caching like this:
```js
this.lastDriverResponse  = { statusId, response };
//...
if(this.lastDriverResponse && this.lastDriverResponse.statusId === driverStatus.id)
    return lastDriverResponse.response;
```
?